src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/energy/EnergyNet.java

```diff
@@ -280,11 +282,29 @@ private int tickAllCapacitors() {
         int supply = 0;
 
         for (Map.Entry<Location, EnergyNetComponent> entry : capacitors.entrySet()) {
-            supply += entry.getValue().getCharge(entry.getKey());
+            supply = flowSafeAddition(supply, entry.getValue().getCharge(entry.getKey()));
         }
 
         return supply;
     }
+    
+    /**
+     * This detects if 2 integers will overflow/underflow and if they will, returns the corresponding value
+     * @param i1 the first integer
+     * @param i2 the second integer
+     * @return {@link Integer#MAX_VALUE} if overflow detected, {@link Integer#MIN_VALUE} if underflow detected, otherwise the sum of i1 and i2
+     */
+    private int flowSafeAddition(int i1, int i2) {
+        boolean willOverflow = (i1 == Integer.MAX_VALUE && i2 > 0 || i2 == Integer.MAX_VALUE && i1 > 0) || i1 > 0 && i2 > Integer.MAX_VALUE - i1;
+        boolean willUnderflow = (i1 == Integer.MIN_VALUE && i2 < 0 || i2 == Integer.MIN_VALUE && i1 < 0) || i1 < 0 && i2 < Integer.MIN_VALUE - i1;
+        if (willOverflow) {
+            return Integer.MAX_VALUE;
+        } else if (willUnderflow) {
+            return Integer.MIN_VALUE;
+        } else {
+            return i1 + i2;
+        }
+    }
```

Comment:
This can probably to go the `NumberUtils` class

Reply:
Also to allow a custom limit in the future, could you add `min` and `max` parameters to this method as well as an overload without them where it would just pass `Integer.MAX_VALUE` and `Integer.MIN_VALUE` as defaults?

Reply:
Ah sure didn't realize we had that class!